os/fs/smartfs/smartfs_utils.c

```diff
@@ -1268,6 +1268,206 @@ int smartfs_createentry(struct smartfs_mountpt_s *fs, uint16_t parentdirsector,
 	return ret;
 }
 
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
+/****************************************************************************
+ * Name: smartfs_findvalidentry
+ * 
+ * Description: Finds an invalid entry in the parent directory that can be
+ *              used to create a new file entry without allocating a new
+ *              sector.
+ *
+ ****************************************************************************/
+
+bool smartfs_findinvalidentry(struct smartfs_mountpt_s *fs, uint16_t parentdirsector, struct smartfs_entry_s *invalidentry)
```

Comment:
Please change bool to int and please see comment below.

Reply:
Agreed